src/pybind/mgr/smb/handler.py

```diff
@@ -473,7 +477,12 @@ def _check(
             elif isinstance(
                 resource, (resources.Share, resources.RemovedShare)
             ):
-                _check_share(resource, staging, self._path_resolver)
+                _check_share(
+                    resource,
+                    staging,
+                    self._path_resolver,
+                    self._earmark_resolver,
```

Comment:
I kinda think path resolver and earmark resolver might be combined (or at least stacked together).